### ISSUE TITLE
APIC Explorer does not generate data properly for model with required date

### ISSUE THREAD

**Issue by samt** [strongloop/loopback-connector-cloudant#23]
A customer wrote into support with the following issue:

> Using APIC, I create a loopback relation from two models, Item and Review.  Item has three fields (id:string, name:string and desc:string).  Review has three fields (reviewer: string, reviewDate: date and review:string).  All three fields are required. For my relationship, Item has many Reviews.   I should also mention that I am using a Cloudant datasource for this scenario. I then go to "Explore" and I use Post to create an Item.  So far everything is fine.  Then, I go to create a "Review" for my Item, here is where things get a little sticky.  First I enter a PersistedModel id for the Item that I want to Post a review for.  Then I click the Generate link to generate the data for the review.  When I do this however, under data the only thing that gets generated is ""  (versus the JSON object that I expect).  If I change my Review model where the reviewDate field is not required, then I see a JSON object displayed, although the behavior still appears to be flaky at times.   I believe there is a problem with the Explore generate data option.

**Comment by jannyHou** [comment]
@samt IIRC apic disabled explorer, are you using **editor**? If it is explorer, could you explain a bit about how to launch apic explorer? Thanks.

**Comment by samt** [comment]
It's in the editor launched with `$ apic edit` in the project directory.

**Comment by jannyHou** [comment]
I created an app according to your description:
https://github.ibm.com/juehou/sandbox

But I didn't see same error happen.
I tried both memory db and cloudant db, they both work fine.

I installed my apiconnect from `https://registry.npmjs.org/`
```
jannyhous-mbp:sandbox jannyhou$ apic -v
API Connect: v5.0.3.0 iFix 2
```

And BTW, honestly I am not quite familiar with apic debugging, like which version should we test and environment setup things. Maybe we can find someone from apic-dev-team to take a look. He/She can simply use my sandbox to test. Just replace cloudant config in `server/datasources.json` with a valid cloudant url.
